discovery/support/chaincode/support_test.go

```diff
@@ -9,66 +9,85 @@ package chaincode
-	ccmd1 := &chaincode.Metadata{Policy: emptySignaturePolicyEnvelope}
-	notEmptySignaturePolicyEnvelope, _ := proto.Marshal(&common.SignaturePolicyEnvelope{
+	emptySignaturePolicyEnvelope := &common.SignaturePolicyEnvelope{}
+	ccmd1 := &chaincode.Metadata{Policy: protoutil.MarshalOrPanic(emptySignaturePolicyEnvelope)}
+	notEmptySignaturePolicyEnvelope := &common.SignaturePolicyEnvelope{
 		Rule:       &common.SignaturePolicy{},
-		Identities: []*msp.MSPPrincipal{{}},
-	})
-	ccmd2 := &chaincode.Metadata{Policy: notEmptySignaturePolicyEnvelope}
+		Identities: []*msp.MSPPrincipal{{Principal: []byte("principal-1")}},
+	}
+	ccmd2 := &chaincode.Metadata{Policy: protoutil.MarshalOrPanic(notEmptySignaturePolicyEnvelope)}
+	notEmptySignaturePolicyEnvelope2 := &common.SignaturePolicyEnvelope{
+		Rule:       &common.SignaturePolicy{},
+		Identities: []*msp.MSPPrincipal{{Principal: []byte("principal-2")}},
+	}
+	ccmd3 := &chaincode.Metadata{Policy: protoutil.MarshalOrPanic(notEmptySignaturePolicyEnvelope),
+		CollectionPolicies: map[string][]byte{"col1": protoutil.MarshalOrPanic(notEmptySignaturePolicyEnvelope2)}}
 
 	tests := []struct {
-		name        string
-		input       *chaincode.Metadata
-		shouldBeNil bool
+		name           string
+		input          *chaincode.Metadata
+		expectedReturn []policies.InquireablePolicy
 	}{
 		{
-			name:        "Nil instantiatedChaincode",
-			input:       nil,
-			shouldBeNil: true,
+			name:           "Nil instantiatedChaincode",
+			input:          nil,
+			expectedReturn: nil,
 		},
 		{
-			name:        "Invalid policy bytes",
-			input:       &chaincode.Metadata{Policy: []byte{1, 2, 3}},
-			shouldBeNil: true,
+			name:           "Invalid policy bytes",
+			input:          &chaincode.Metadata{Policy: []byte{1, 2, 3}},
+			expectedReturn: nil,
 		},
 		{
-			name:        "Empty signature policy envelope",
-			input:       ccmd1,
-			shouldBeNil: true,
+			name:           "Empty signature policy envelope",
+			input:          ccmd1,
+			expectedReturn: nil,
 		},
 		{
-			name:        "Not Empty signature policy envelope",
-			input:       ccmd2,
-			shouldBeNil: false,
+			name:           "Not Empty signature policy envelope",
+			input:          ccmd2,
+			expectedReturn: []policies.InquireablePolicy{inquire.NewInquireableSignaturePolicy(notEmptySignaturePolicyEnvelope)},
+		},
+		{
+			name:           "Not Empty signature policy envelopes with existing collection policy",
+			input:          ccmd3,
+			expectedReturn: []policies.InquireablePolicy{inquire.NewInquireableSignaturePolicy(notEmptySignaturePolicyEnvelope2)},
 		},
 	}
 
 	for _, test := range tests {
 		test := test
 		t.Run(test.name, func(t *testing.T) {
 			sup := NewDiscoverySupport(&mockMetadataRetriever{res: test.input})
-			res := sup.PoliciesByChaincode("", "")
-			if test.shouldBeNil {
+			var res []policies.InquireablePolicy
+			if test.input == ccmd3 {
+				res = sup.PoliciesByChaincode("", "", "col1")
```

Comment:
let's please extract this variable ("col1") to the test parameters and not use  **if test.input == ccmd3** :)

Reply:
Done.